CONTRIBUTING.md

```diff
@@ -1,10 +1,40 @@
-You want to contribute? We really appreciate that.
-Please have a look at this list to facilitate your contribution:
+# Want to contribute some code? 
+
+## We’d really appreciate your support!
+
+- If the [Github issue](https://github.com/jamulussoftware/jamulus/issues) for your feature/bug fix already exists, write a message in that issue saying you want to work on it. 
+- Otherwise, please [post on the GitHub Discussions](https://github.com/jamulussoftware/jamulus/discussions) and say that you are planning to do some coding and explain why. Then we can discuss the specification. 
+- Please only start coding after we have agreed to a specification.
+
+## Jamulus project/source code general principles
+
+### Stability
+This has been, and must continue to be, the most important requirement. It is particularly important for use cases such as the Worldjam where instability during a live performance would be unacceptable. The following two principles are designed to support this principle.
+
+### [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
+If a feature or function can be achieved in another way by another system or method, then that is preferable to building that feature into Jamulus. 
+
+### Focus on and improve the strengths of Jamulus
+There are other solutions for real time online jamming available. But Jamulus stands alone with its client/server system, minimalistic GUI and simple functionality. 
+
+### Source code consistency
+- Respect the existing code style: Tab size=4, insert spaces.
+- Space before and after `(` and `)`, except no space between `)` and `;`, and no space before an empty `()`.
+- All bodies of `if`, `else`, `while`, `for`, etc., to be enclosed in braces `{` and `}`, usually on separate lines.
+
+Make sure (if possible) that your code compiles on Windows/Mac/Linux.
+Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
+
+### Documentation/Acknowledgements
+Each new feature or bug fix must be documented in the ChangeLog. If a new contributor/translator adds code to the Jamulus project, he/she should be added to the contributor/translator list in the About dialog of the Jamulus software (see in `src/util.cpp` in the constructor function `CAboutDlg::CAboutDlg()`).
```

Comment:
For 3.7.0 we asked contributors to create the Changelog entry. This has the downside of creating conflicts, which is annoying for other contributors. I suggest to change this to have people optionally suggest a changelog entry line in the PR description, but create the actual Changelog entries as part of the release process.

I think we can stick to people adding themselves to the contributors list because this is a one-time action per-person.

I also suggest to make it clearer whose job it is to do these things.

Suggestion:
```suggestion
Each new feature or bug fix must be documented in the ChangeLog. Please provide a single-sentence suggestion for that as part of your pull request description. Do not modify the ChangeLog file as part of your PR as it will lead to conflicts.

If you are a first-time contributor/translator, please add yourself to the contributor/translator list in the About dialog of the Jamulus software (see in `src/util.cpp` in the constructor function `CAboutDlg::CAboutDlg()`).
```